boa3/analyser/typeanalyser.py

```diff
@@ -705,22 +705,42 @@ def _get_is_instance_function_calls(self, node: ast.AST) -> Tuple[Dict[str, ISym
                 condition = condition.operand
                 negate = True
 
-            if (isinstance(condition, ast.Call) and isinstance(condition.func, ast.Name)
-                    and condition.func.id in is_instance_objs and len(condition.args) == 2):
-                original = self.get_symbol(condition.args[0])
-                if isinstance(original, Variable) and isinstance(condition.args[0], ast.Name):
-                    original_id = condition.args[0].id
+            # verifies if condition is an is_instance condition
+            is_instance_condition = (isinstance(condition, ast.Call) and isinstance(condition.func, ast.Name)
+                                     and condition.func.id in is_instance_objs and len(condition.args) == 2)
+
+            # verifies if condition is a identity condition (is None or is not None)
+            from boa3.model.operation.unary.noneidentity import NoneIdentity
+            from boa3.model.operation.unary.nonenotidentity import NoneNotIdentity
+            identity_condition = isinstance(condition, ast.Compare) \
+                                 and isinstance(condition.ops[0], (NoneIdentity, NoneNotIdentity))
+
+            if identity_condition and isinstance(condition.ops[0], NoneNotIdentity):
+                negate = True
```

Comment:
```suggestion
            identity_condition = isinstance(condition, ast.Compare) \
                                 and condition.ops[0] in (BinaryOp.IsNone, BinaryOp.IsNotNone))

            if identity_condition and condition.ops[0] is BinaryOp.IsNotNone:
                negate = True
```

Doesn't this work?
Does it need to check the type of the object or just validating if it's the same object defined in the BinaryOp is enough?

Reply:
It doesn't, it would need to be: 
```python
            identity_condition = isinstance(condition, ast.Compare) \
                                 and type(condition.ops[0]) in (type(BinaryOp.IsNone), type(BinaryOp.IsNotNone))
            if identity_condition and type(condition.ops[0]) is type(BinaryOp.IsNotNone):
                negate = True
```

Reply:
I see.
I think it's better to use `type(BinaryOp.IsNone)` and `type(BinaryOp.IsNotNone)`, since this is used a few instructions after for `Builtin.IsInstance`